src/server.py

```diff
@@ -1,16 +1,32 @@
 import asyncio
 import asyncpg
+import secrets
+import string
 import random
+import time
+import json
+from datetime import datetime, timedelta
 from contextlib import asynccontextmanager
 from fastapi import FastAPI, HTTPException
 from fastapi.staticfiles import StaticFiles
 from deck import Deck
 
 SUITS = ["C", "D", "H", "S"]
-RANKS = ["2", "3", "4", "5", "6", "7", "8", "9", "0", "J", "Q", "K", "A"]
+RANKS = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"]
 
 app = FastAPI()
 
+# Dictionaries to store client IDs and associated objects
+deck_objects: dict[str, Deck] = {}
+expiration_times: dict[str, float] = {}
+
+def generate_unique_deck_id(length=20):
+    alphabet = string.ascii_letters + string.digits
+    while True:
+        deck_id = ''.join(secrets.choice(alphabet) for _ in range(length))
```

Comment:
Interesting, is this secure? How many possibilities are there? What's the probability of a random collision if someone is trying to generate ids over and over?

Reply:
There are 62 possible symbols and 20 characters each, giving $7 \times 10^{35}$ possibilities. Call this quantity $N$ for now. For the possibility of a collision, this is just the birthday problem. The probability depends on the number of IDs generated, but using <https://www.bdayprob.com/> (because I don't feel like doing that math manually), if we're to accept a probability of $10^{-15}$ of collision with this space, then we would need to see around $2^{35}$ or 34 million decks created before this might happen.

Secure enough for our purposes, certainly, but I'm not sure about if we want more than that many users.

Reply:
Excellent analysis :)